### PR TITLE
Add frame length, cycle time & extended property to header output of generate_c_source

### PR DESCRIPTION
**Pull request by Adraub**
Add these properties to header in order to be able to exploit them in C
Update test headers with new outputs

### CODE REVIEW

File: tests/files/c_source/abs.h [hunk 1]
```diff
@@ -58,6 +58,63 @@
 #define ABS_BREMSE_50_FRAME_ID (0x560u)
 #define ABS_BREMSE_53_FRAME_ID (0x5c0u)
 
+#define ABS_BREMSE_33_FRAME_LEN (8u)
```
Comment (eerimoq) [inline tests/files/c_source/abs.h]: I suggest `#define ABS_BREMSE_33_LENGTH (8u)` to have the same naming as the database message class.

File: tests/files/c_source/abs.h [hunk 2]
```diff
@@ -58,6 +58,63 @@
...
+#define ABS_ABS_SWITCH_FRAME_LEN (8u)
+#define ABS_BREMSE_30_FRAME_LEN (8u)
+#define ABS_BREMSE_31_FRAME_LEN (8u)
+#define ABS_BREMSE_32_FRAME_LEN (8u)
+#define ABS_BREMSE_51_FRAME_LEN (8u)
+#define ABS_BREMSE_52_FRAME_LEN (8u)
+#define ABS_BREMSE_50_FRAME_LEN (8u)
+#define ABS_BREMSE_53_FRAME_LEN (8u)
+
+#define ABS_BREMSE_33_FRAME_EXTENDED (0)
```
Comment (eerimoq) [inline tests/files/c_source/abs.h]: I suggest `#define ABS_BREMSE_33_IS_EXTENDED (0)` to have the same naming as the database message class.

File: tests/files/c_source/abs.h [hunk 3]
```diff
@@ -58,6 +58,63 @@
...
+#define ABS_ABS_SWITCH_FRAME_EXTENDED (0)
+#define ABS_BREMSE_30_FRAME_EXTENDED (0)
+#define ABS_BREMSE_31_FRAME_EXTENDED (0)
+#define ABS_BREMSE_32_FRAME_EXTENDED (0)
+#define ABS_BREMSE_51_FRAME_EXTENDED (0)
+#define ABS_BREMSE_52_FRAME_EXTENDED (0)
+#define ABS_BREMSE_50_FRAME_EXTENDED (0)
+#define ABS_BREMSE_53_FRAME_EXTENDED (0)
+
+#define ABS_BREMSE_33_FRAME_CYCLE_MS (10u)
```
Comment (eerimoq) [inline tests/files/c_source/abs.h]: I suggest `#define ABS_BREMSE_33_CYCLE_TIME_MS (10u)` to have the same naming as the database message class. `_MS` suffix is a good addition for clarity.

File: cantools/database/can/c_source.py [hunk 4]
```diff
@@ -1250,6 +1253,39 @@ def _generate_frame_id_defines(database_name, messages):
...
 
+def _generate_frame_length_defines(database_name, messages):
+    result = '\n'.join([
+        '#define {}_{}_FRAME_LEN ({}u)'.format(
+            database_name.upper(),
+            message.snake_name.upper(),
+            message.length)
+        for message in messages
+    ])
+    return '\n'+result if result else ''
```
Comment (eerimoq) [inline cantools/database/can/c_source.py]: Add a blank line before the return.

File: cantools/database/can/c_source.py [hunk 5]
```diff
@@ -1250,6 +1253,39 @@ def _generate_frame_id_defines(database_name, messages):
...
+
+def _generate_frame_cycle_defines(database_name, messages):
+    result = '\n'.join([
+        '#define {}_{}_FRAME_CYCLE_MS ({}u)'.format(
+            database_name.upper(),
+            message.snake_name.upper(),
+            message.cycle_time)
+        for message in messages if message.cycle_time
+    ])
+    return '\n'+result+'\n' if result else ''
```
Comment (eerimoq) [inline cantools/database/can/c_source.py]: Add a blank line before the return.

File: cantools/database/can/c_source.py [hunk 6]
```diff
@@ -1250,6 +1253,39 @@ def _generate_frame_id_defines(database_name, messages):
...
+    return '\n'+result if result else ''
+
+
+def _generate_frame_cycle_defines(database_name, messages):
+    result = '\n'.join([
+        '#define {}_{}_FRAME_CYCLE_MS ({}u)'.format(
+            database_name.upper(),
+            message.snake_name.upper(),
+            message.cycle_time)
+        for message in messages if message.cycle_time
```
Comment (eerimoq) [inline cantools/database/can/c_source.py]: Should be `message.cycle_time is not None`.

**Comment by Adraub** [comment]
I applied suggested comments thanks for the review

**Comment by eerimoq** [comment]
No problem, thanks for the PR!

**Comment by eerimoq** [comment]
Part of version 32.17.0, available on PyPI!

**Pull request merged**